### PR TITLE
Step 4: add support for unique with dim=None

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #18661 Step 7: remove _unique
* #18655 Step 6: Rename _unique2 to unique and add int? dim
* #18654 Step 5: remove _unique_dim in favor of unique_dim
* **#18651 Step 4: add support for unique with dim=None**

See step 1 for the description of the full stack

Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D15000463](https://our.internmc.facebook.com/intern/diff/D15000463)

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
@VitalyFedyunin step 4 and 5 should also be ready

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -1929,7 +1929,8 @@
 # the below operator is a temporary hack for adding return_counts support
 # Please don't rely on these two operators, they will be removed soon
 
-- func: _unique2_temporary_will_remove_soon(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False) -> (Tensor, Tensor, Tensor)
+- func: _unique2(Tensor self, bool sorted=True, bool return_inverse=False, bool return_counts=False) -> (Tensor, Tensor, Tensor)
+  matches_jit_signature: True
```
Comment (VitalyFedyunin) [inline aten/src/ATen/native/native_functions.yaml]: This is now default value, please remove

**Review by VitalyFedyunin — APPROVED** [review]

**Comment by VitalyFedyunin** [comment]
Please rebase upon master it will help to import changes.

**Comment by zasdfgbnm** [comment]
@VitalyFedyunin All fixed

**Comment by VitalyFedyunin** [comment]
Thanks! There are couple tests I'm still waiting to finish and I will land afterwards.

**Pull request closed without merging**